sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/QueryStageExec.scala

```diff
@@ -219,11 +217,15 @@ case class ReusedQueryStageExec(
     }
   }
 
-  override def outputPartitioning: Partitioning = plan.outputPartitioning match {
+  private[sql] def updatePartitioning(p: Partitioning): Partitioning = p match {
```

Comment:
nit: Is there a better place to put this method in, as well as `updateAttr`, like in `Attribute` or sth., so that they can be shared between `ReusedExchangeExec` and `ReusedQueryStageExec`?